src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -303,6 +304,37 @@ static bool can_reshape_be_optimized(const reshape_node& node) {
     return node.is_in_place() && node.get_fused_activations_funcs().empty();
 }
 
+#ifdef ENABLE_ONEDNN_FOR_GPU
+// pattern #1: input -> (bfyx) -> permute -> (byxf) -> oneDNN convolution
+//      - permute node reorders is [0, 3, 1, 2] (bfyx -> byxf).
+//      - input layout of onednn convolution is byxf.
+// pattern #2: oneDNN convolution convolution -> (byxf) -> permute -> (bfyx) -> output
+//      - permute node reorders is [0, 2, 3, 1] (byxf -> bfyx).
+//      - output layout of permute is byxf
+static bool can_permute_be_optimized(const permute_node& node) {
+    if (node.is_output() || node.get_dependencies().size() != 1 || node.get_users().size() != 1
+        || !node.get_output_layout().is_static()
+        || !node.get_dependency(0).get_output_layout().is_static())
+        return false;
+
+    if (node.get_output_layout().format != cldnn::format::byxf
+        && node.get_output_layout().format != cldnn::format::bfyx)
+        return false;
+
+    // permute - convolution
+    auto next = node.get_users().front();
+    if (next->is_type<convolution>() && next->get_preferred_impl_type() == impl_types::onednn)
+        return node.is_reverse_rotating_except_batch();
+
+    // convolution - permute
+    auto& prev = node.get_dependency(0);
+    if (prev.is_type<convolution>() && prev.get_preferred_impl_type() == impl_types::onednn)
+        return node.is_rotating_except_batch();
```

Comment:
in `conv-permute` pattern, permute output is also bfyx/byxf? I think it might be some blocked format...

Reply:
In this pattern, the output is byxf and the input might be blocked format. The output format should be byxf so that the next permute will be optimized out. What am I missing?

Reply:
Hmm I think I was confused. Let me close this comment.

Reply:
What about the pattern like this? (copied from your note) In this case, permute is generating blocked format. And from your code, it won't be optimized out because permute output is not simple format?

|  123 | conv2d_109/Conv2D                     | Convolution | 163          | [122]            | [32, 1024, 13, 13]  | bs_fs_yx_b32_f32 | I8       | [32, 255, 13, 13]   | bs_fs_yx_b32_f16 | FP16     | jit:ir__i8                       |
|  124 | conv2d_109/BiasAdd/Add                | Permute     | 136          | [123]            | [32, 255, 13, 13]   | bs_fs_yx_b32_f16 | FP16     | [32, 13, 13, 255]   | bs_fs_yx_b32_f16 | FP16     | permute_ref__f16                 |
|  125 | conv2d_109/BiasAdd:0                  | Reorder     | 747          | [124]            | [32, 13, 13, 255]   | bs_fs_yx_b32_f16 | FP16     | [32, 13, 13, 255]   | bfyx             | FP32     | reorder_data__f32                |

Reply:
This code is removed.